packages/css/fieldset.css

```diff
@@ -1,43 +1,43 @@
 .ds-fieldset {
+  --dsc-fieldset-icon-size: 1.2em;
+  --dsc-fieldset-gap: var(--ds-spacing-2);
+
   margin: 0;
   padding: 0;
   border: 0;
   min-width: 0;
-}
-
-.ds-fieldset--spacing {
-  display: flex;
-  flex-direction: column;
-  gap: var(--ds-spacing-2);
-}
-
-.ds-fieldset__description {
-  color: var(--ds-color-neutral-text-subtle);
-  margin-bottom: var(--ds-spacing-2);
-}
-
-.ds-fieldset__legend {
-  display: contents;
-}
-
-.ds-fieldset--readonly .ds-fieldset__legend__content {
-  display: inline-flex;
-}
-
-.ds-fieldset:not(:has(.ds-fieldset__description)) .ds-fieldset__legend__content {
-  margin-bottom: var(--ds-spacing-2);
-}
-
-.ds-fieldset--readonly .ds-fieldset__readonly__icon {
-  height: 1.2em;
-  width: 1.2em;
-}
-
-.ds-fieldset:disabled .ds-fieldset__legend,
-.ds-fieldset:disabled .ds-fieldset__description {
-  color: var(--ds-color-neutral-border-subtle);
-}
 
-.ds-fieldset__error-message {
-  display: contents;
+  & > :not(:last-child) {
+    margin-bottom: var(--ds-spacing-2); /* Use margin as fieldset does not play nice with display: flex */
+  }
+
+  & > legend {
+    display: inline-flex;
+  }
+
+  & > legend:empty {
+    display: none;
+  }
+
+  &[data-hidelegend] > legend {
+    @composes ds-sr-only from '../css/utilities.css';
+  }
```

Comment:
This does not hide the description, as it does currently.
Should this change maybe be done in another PR?

Reply:
oh right! Sorry was experimenting and forgot to bring back. I'll fix, well spotted ☺️ 
I think we should just have in the same PR 👍

Reply:
If we agree to keep description when legend is hidden, I'm all for keeping it in this PR - but that might take some time to decide on